tests/tests_integration/test_api/test_datapoints.py

```diff
@@ -992,6 +1023,361 @@ def test_query_with_duplicates(self, retrieve_endpoints, one_mill_dps_ts, ms_bur
+            ("119", "2023-01-01T00:00:00+00:00", "2023-01-02T00:00:01+00:00", "sum", "5h"),
+            ("119", "2023-01-01T00:00:00+00:00", "2023-01-02T00:00:01+00:00", "count", "5h"),
+            ("120", "2023-01-01T00:00:00+00:00", "2023-01-02T00:00:59+00:00", "average", "2m"),
+            ("120", "2023-01-01T00:00:00+00:00", "2023-01-02T00:00:01+00:00", "sum", "30m"),
+            ("120", "2023-01-01T00:00:00+00:00", "2023-01-01T23:59:01+00:00", "average", "15m"),
+            ("120", "2023-01-01T00:00:00+00:00", "2023-01-01T23:59:01+00:00", "average", "1h"),
+            ("120", "2023-01-01T00:00:00+00:00", "2023-01-01T23:59:01+00:00", "count", "38m"),
+        ),
+    )
+    def test_cdf_aggregate_equal_to_cdf(
+        test_series_no: str,
+        start: str,
+        end: str,
+        aggregation: Literal["average", "sum", "count"],
+        granularity: str,
+        cognite_client: CogniteClient,
+        all_test_time_series: TimeSeriesList,
+    ):
+        # Arrange
+        time_series = get_test_series(test_series_no, all_test_time_series)
+        start, end = datetime.fromisoformat(start), datetime.fromisoformat(end)
+        raw_df = cognite_client.time_series.data.retrieve_dataframe(
+            external_id=time_series.external_id, start=start, end=end
+        )
+        expected_aggregate = cognite_client.time_series.data.retrieve_dataframe(
+            start=start,
+            end=end,
+            external_id=time_series.external_id,
+            aggregates=aggregation,
+            granularity=granularity,
+            include_aggregate_name=False,
+            include_granularity_name=False,
+        )
+
+        # Act
+        actual_aggregate = cdf_aggregate(raw_df, aggregation, granularity, time_series.is_step)
+
+        # Assert
+        # Pandas adds the correct frequency to the index, while the SDK does not when uniform is not True.
+        # The last point is not compared as the raw data might be missing information to do the correct aggregate.
+        pd.testing.assert_frame_equal(
+            expected_aggregate.iloc[:-1], actual_aggregate.iloc[:-1], check_freq=False, check_exact=False
+        )
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "aggregation, granularity, tz_name",
+        (
+            ("average", "1d", "Europe/Oslo"),
+            ("average", "5d", "Europe/Oslo"),
+            ("sum", "31d", "Europe/Oslo"),
+        ),
+    )
+    def test_retrieve_dataframe_in_tz_aggregate_raw_hourly(
+        aggregation: Literal["average", "sum"],
+        granularity: str,
+        tz_name: str,
+        cognite_client: CogniteClient,
+        hourly_2023: pd.DataFrame,
+    ):
+        # Arrange
+        tz = ZoneInfo(tz_name)
+        start = datetime(2023, 1, 1, tzinfo=tz)
+        end = datetime(2023, 12, 31, 23, 0, 0, tzinfo=tz)
+        raw_df = hourly_2023.tz_convert(tz_name).loc[start:end].copy()
```

Comment:
Avoid ignored (but still warned) AttributeError from pandas cinit (doesn't like ZoneInfo in start/end). Could make an additional pd_start and pd_end, but a clever hack is just to make them strings:
```suggestion
        raw_df = hourly_2023.tz_convert(tz_name).loc[str(start):str(end)].copy()
```